packages/ra-ui-materialui/src/input/AutocompleteInput.js

```diff
@@ -455,6 +462,7 @@ export class AutocompleteInput extends React.Component {
             options,
         } = this.props;
         const { suggestions, searchText } = this.state;
+        const slicedSuggestions = this.sliceSuggestions(suggestions);
```

Comment:
The result will be a new array for each `render()`, which will force a rerender of the autosuggest even if the suggestions didn't change. This is bad, so you should slice when setting the suggestions in the state, not when reading the state.